### PR TITLE
Fix rx arping socket bind

### PR DESCRIPTION
**Pull request by thattil**
- popen to execute arping was forking so it maintained that the socket was open (if it was forked when the udpsocket was open) when it was not anymore in a scropt of start and stop reciever. So Forking before calling popen and then moving delete udp pointer to after the listening threads are done running so that rx_arping cannot be set in between an acquiistion or when udp socket is already open.

### CODE REVIEW

File: slsReceiverSoftware/src/Listener.cpp [hunk 1]
```diff
@@ -184,6 +184,14 @@ void Listener::ShutDownUDPSocket() {
     }
 }
 
+void Listener::DeleteUDPSocket() {
+    if (udpSocket) {
+        udpSocket = nullptr;
```
Comment (erikfrojdh) [inline slsReceiverSoftware/src/Listener.cpp:189]: udpSocket.reset() might be more clear but would to the same thing.

**Review by erikfrojdh — COMMENTED** [review]
Remove loop otherwise looks ok

File: slsReceiverSoftware/src/Arping.cpp [hunk 2]
```diff
@@ -33,59 +35,68 @@ void Arping::SetInterfacesAndIps(const int index, const std::string &interface,
...
+        // just print (was already tested at Process start)
         if (!error.empty()) {
             LOG(logERROR) << error;
         }
 
-        // wait for 60s as long as thread not killed
+        // wait for 60s as long as Process not killed
         int nsecs = 0;
-        while (runningFlag && nsecs != 60) {
+        while (nsecs != timeIntervalSeconds) {
```
[the remark this answers is not in the document]
Comment (thattil) [reply slsReceiverSoftware/src/Arping.cpp:88]: ah yes, not anymore. will fix

File: slsReceiverSoftware/src/Listener.cpp (hunk 1 above)
Comment (thattil) [reply slsReceiverSoftware/src/Listener.cpp:189]: done

**Pull request merged**